### ISSUE TITLE
IPojoComponentFeature not working in the latest version (jadex-3.0.113)

### ISSUE THREAD

**Issue by daniel-souza** [actoron/jadex#1]
A defined POJO Agent with correctly getters and setters currently can only be accessed by injection Inside a **Service:**

    @ServiceComponent
    POJOAgent pojoAgent;

**Following the tutorial (Docs » Active Components Tutorial » External Access), I cannot perform the following instruction**:

    public class BotGuiF3 extends AbstractComponentViewerPanel {
		@Override
		public JComponent getComponent() {
			final BotPanel botPanel = new BotPanel();
			final IExternalAccess agent = getActiveComponent();
			IFuture<String[]> fut = agent.scheduleStep(new IComponentStep<String[]>() {
			
				//@ServiceComponent
				//ChatBotF2Agent chatbot; // this injection not happens, since it is not a service procedure.

				public IFuture<String[]> execute(IInternalAccess ia) {
					ChatBotF3Agent chatbot = (ChatBotF3Agent)((IPojoComponentFeature)ia).getPojoAgent();
					return new Future<String[]>(new String[] {chatbot.getKeyword(), chatbot.getReply()});
				}
			
			});

			fut.addResultListener(new SwingDefaultResultListener<String[]>() {
			
				@Override
				public void customResultAvailable(String[] result) {
					botPanel.getKeywordTextField().setText(result[0]);
					botPanel.getReplyTextField().setText(result[1]);
				}
			
			});
			
			...
	}

When executing the following line `ChatBotF3Agent chatbot = (ChatBotF3Agent)((IPojoComponentFeature)ia).getPojoAgent();` Throws:

```
java.lang.ClassCastException: class jadex.platform.service.cms.PlatformComponent cannot be cast to class jadex.bridge.component.IPojoComponentFeature (jadex.platform.service.cms.PlatformComponent and jadex.bridge.component.IPojoComponentFeature are in unnamed module of loader 'app')
	at tutorial.chatservice.BotGuiF3$1.execute(BotGuiF3.java:32)
	at jadex.bridge.component.impl.ExecutionComponentFeature.execute(ExecutionComponentFeature.java:1176)
	at jadex.commons.concurrent.Executor.code(Executor.java:299)
	at jadex.commons.concurrent.Executor.run(Executor.java:126)
	at jadex.platform.service.execution.AsyncExecutionService$1.run(AsyncExecutionService.java:109)
	at jadex.commons.concurrent.ThreadPool$ServiceThread.run(ThreadPool.java:515)

Dec 21, 2018 11:54:47 PM jadex.commons.future.DefaultResultListener exceptionOccurred
WARNING: Exception occurred: tutorial.chatservice.BotGuiF3$2@410f4d40, java.lang.ClassCastException: class jadex.platform.service.cms.PlatformComponent cannot be cast to class jadex.bridge.component.IPojoComponentFeature (jadex.platform.service.cms.PlatformComponent and jadex.bridge.component.IPojoComponentFeature are in unnamed module of loader 'app')
```

`IPojoComponentFeature are in unnamed module of loader 'app'`

**Java Environment that reproduced this problem**:

```
openjdk version "11.0.1" 2018-10-16
OpenJDK Runtime Environment (build 11.0.1+13-suse-lp150.2.9.1-x8664)
OpenJDK 64-Bit Server VM (build 11.0.1+13-suse-lp150.2.9.1-x8664, mixed mode)
```
The tutorial points to `(ChatBotF3Agent)((IPojoMicroAgent)ia).getPojoAgent();`, whose `IPojoMicroAgent` no longer exists in this Jadex version.

**Comment by daniel-souza** [comment]
Checking the code I found how to get the POJOAgent, I need to do the following:
```
MicroPojoComponentFeature pojo = new MicroPojoComponentFeature(ia, null);
HelloAgent agent = (HelloAgent)pojo.getPojoAgent();
```
I don't know deeply the class hierarchy, but It could be easier than instantiating a new MicroPojoComponentFeature.

To only get the POJO, the second parameter `ComponentCreationInfo` is never used.

Then, I'd suggest to do the following, as implemented inside the constructor of MicroPojoComponentFeature:
```
static public Object getPojo(IInternalAccess ia) {
	try {
		// Create the pojo agent
		MicroModel model = (MicroModel)getComponent().getModel().getRawModel();
		return model.getPojoClass().getType(model.getClassloader()).newInstance();
	}
	catch(Exception e)
	{
		throw new RuntimeException(e);
	}
	return null;
}
```
Further, without need of the `MicroPojoComponentFeature` instantiation, we could get the pojo like that:
```
POJOAgent pojoAgent = (POJOAgent)MicroPojoComponentFeature.getPojo(ia);
```

**Comment by kj-actoron** [comment]
Thank you for trying Jadex, unfortunately the tutorial of version 3.x on the website is lagging a bit behind and reflects the state before the introduction of component features.

Features modularize component functionality, so casting the IInternalAccess is not the correct way for accessing component features, which should be acquired using the getComponentFeature() (Jadex 3.x) or getFeature() (Jadex 4.x) method:

`ChatBotF3Agent  chatbot = ChatBotF3Agent)ia.getComponentFeature(IPojoComponentFeature.class).getPojoAgent();`

Until we fix the website, updated (and working) tutorial source code can be found in the jadex-applications-micro subproject, including the F3 example.

**Comment by kj-actoron** [comment]
We've now released Jadex 3.0.115 and with it the error in the tutorial. Thank you for pointing it out to us.
